R/sits_kohonen.R

```diff
@@ -1,92 +1,770 @@
+                    device = "png"
+                )
+            }
+
+            #-------------------------------------------------------------------------------
+            #get samples of current group
+            s = 1
+            s_group <- as_tibble()
+            for (s in 1:length(neuron_id_int))
+            {
+                temporary_samples <-
+                    dplyr::filter(current_class, current_class$id_neuron == neuron_id_int[s])
+                s_group <- rbind(s_group, temporary_samples)
+            }
+            s_group_cluster <- s_group
+            s_group_cluster$label2 <-paste(class_neurons[k], "_", j,  sep = '')
+
+            Class_group <-rbind(Class_group, s_group_cluster)
+        }
+    }
+    return(Class_group)
+}
+
+#' @title Metrics by cluster
+#' @name sits_metrics_by_cluster
+#' @author Lorena Santos, \email{lorena.santos@@inpe.br}
+#'
+#' @description This function extracts metrics about the clusters calculating
+#' the percentage of mixture between a cluster and others.
+#'
+#' @param info_sample_cluster.tb tibble containg information about evaluation of samples
+
+#' @return returns the confusion matrix and a table with percentage of mixture between the clusters
+#' @export
+
+sits_metrics_by_cluster<-function(info_sample_cluster.tb)
+{
+
+    #get only id, label and neuron_label
+    temp.data.tb<-unique(dplyr::select(info_sample_cluster.tb, id_sample, label, neuron_label))
+
+    #get label that no have cluster
+    no_cluster<-dplyr::setdiff(temp.data.tb$label, temp.data.tb$neuron_label)
+
+    confusion.matrix<-table(temp.data.tb$label, temp.data.tb$neuron_label)
+
+    #get the names of classes (original labels from samples)
+    Label_table<-rownames(confusion.matrix)
+
+    if (length(no_cluster)>0){
+
+        #number of class that dont have cluster
+        size_vector<-length(no_cluster)
+
+        #Add columns in confusion matrix
+        sv=1
+        for(sv in 1:size_vector)
+        {
+            #position to add column in confusion matrix
+            positon_to_add<-which(Label_table==no_cluster[sv])
+
+            #trasform in df.array to add a column
+            cf<-as.data.frame.array(confusion.matrix)
+
+            #add a column in same position of line
+            cf<-add_column(cf, d = 0, .after =positon_to_add-1 )
+
+            #rename de column to name of cluster
+            names(cf)[positon_to_add]<-no_cluster[sv]
+            confusion.matrix<-as.table(as.matrix(cf))
+        }
+    }
+
+    #Add the total number of samples in table
+    confusion.matrix.tb <- addmargins(confusion.matrix, FUN = list(Total = sum), quiet = TRUE)
+
+    #get dimensions (rows and col)
+    #represents the original classes of samples
+    dim_row <-dim(confusion.matrix.tb)[1]
+
+    #represents clusters
+    dim_col <-dim(confusion.matrix.tb)[2]
+
+    #get the names of classes (original labels from samples)
+    Label_table<-rownames(confusion.matrix.tb)[1:dim_row-1]
```

Comment:
sits variables and functions always start with lowercase.

Reply:
revised